src/plugins/controls/public/control_group/utils/has_compressed.ts

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+export interface HasCompressed {
+  compressed: boolean;
+}
+
+export const apiHasCompressed = (unknownApi: unknown): unknownApi is HasCompressed => {
```

Comment:
type guards should check for `undefined`. So you guard should look like `Boolean(unknownApi && typeof (unknownApi as HasCompressed).compressed === 'boolean'`